server/jetstream_cluster.go

```diff
@@ -1409,7 +1409,14 @@ func (js *jetStream) monitorCluster() {
 					continue
 				}
 				if didSnap, didStreamRemoval, didConsumerRemoval, err := js.applyMetaEntries(ce.Entries, ru); err == nil {
-					_, nb := n.Applied(ce.Index)
+					var nb uint64
+					js.mu.RLock()
+					shuttingDown := js.shuttingDown
```

Comment:
I don't like that we need this Rlock here.

There is a convenience function here for this already

```
// Helper for determining if we are shutting down.
func (js *jetStream) isShuttingDown() bool {
	js.mu.RLock()
	defer js.mu.RUnlock()
	return js.shuttingDown
}
```

Should we make an atomic?

Reply:
What does applied return for nb when shutting down?

Reply:
Am now using `js.isShuttingDown()` instead.

> What does applied return for nb when shutting down?

It returns the amount of total bytes currently in our WAL. Doesn't take into account if we're shutting down or not.
The important bit is that we should not be changing `n.applied` while `js.processConsumerRemoval` (for example) functions are failing due to `shuttingDown` and then a higher `n.applied` being used in a snapshot during shutdown. Which results in the `js.processConsumerRemoval` then not being replayed of course.